### ISSUE TITLE
More failing tests on Windows (Python 3)

### ISSUE THREAD

**Issue by aldanor** [mitsuhiko/logbook#119]
Looks like the first two are unicode-related, and the third one has to do something with tempfile. The tests fail on both py33 and py34, but pass on py27. Will take a closer look.

```python
_______________ test_file_handler_unicode[activation_strategy0] _______________

logfile = 'C:\\Users\\Smirnov\\AppData\\Local\\Temp\\pytest-63\\test_file_handler_unicode_acti0\\logfile.log'
activation_strategy = <class 'tests.conftest.ContextEnteringStrategy'>
logger = <logbook.base.Logger object at 0x00000000050EFB70>

    def test_file_handler_unicode(logfile, activation_strategy, logger):
        with capturing_stderr_context() as captured:
            with activation_strategy(logbook.FileHandler(logfile)):
                logger.info(u('\u0431'))
>       assert (not captured.getvalue())
E       assert not 'Traceback (most recent call last):\n  File "tests\\test_file_handler.py, line 25\n'
E        +  where 'Traceback (most recent call last):\n  File "tests\\test_file_handler.py, line 25\n' = <built-in method getvalue of _io.StringIO object at 0x0000000004F5D318>()
E        +    where <built-in method getvalue of _io.StringIO object at 0x0000000004F5D318> = <_io.StringIO object at 0x0000000004F5D318>.getvalue

tests\test_file_handler.py:26: AssertionError
```

```python
_______________ test_file_handler_unicode[activation_strategy1] _______________

logfile = 'C:\\Users\\Smirnov\\AppData\\Local\\Temp\\pytest-63\\test_file_handler_unicode_acti1\\logfile.log'
activation_strategy = <class 'tests.conftest.PushingStrategy'>
logger = <logbook.base.Logger object at 0x000000000510A320>

    def test_file_handler_unicode(logfile, activation_strategy, logger):
        with capturing_stderr_context() as captured:
            with activation_strategy(logbook.FileHandler(logfile)):
                logger.info(u('\u0431'))
>       assert (not captured.getvalue())
E       assert not 'Traceback (most recent call last):\n  File "tests\\test_file_handler.py, line 25\n'
E        +  where 'Traceback (most recent call last):\n  File "tests\\test_file_handler.py, line 25\n' = <built-in method getvalue of _io.StringIO object at 0x0000000004FC7048>()
E        +    where <built-in method getvalue of _io.StringIO object at 0x0000000004FC7048> = <_io.StringIO object at 0x0000000004FC7048>.getvalue

tests\test_file_handler.py:26: AssertionError
```

```python
______________________ test_external_application_handler ______________________

tmpdir = local('C:\\Users\\Smirnov\\AppData\\Local\\Temp\\pytest-63\\test_external_application_hand0')
logger = <logbook.base.Logger object at 0x00000000052B10B8>

    def test_external_application_handler(tmpdir, logger):
        from logbook.more import ExternalApplicationHandler as Handler
        fn = tmpdir.join('tempfile')
        handler = Handler([sys.executable, '-c', r'''if 1:
        f = open(%(tempfile)s, 'w')
        try:
            f.write('{record.message}\n')
        finally:
            f.close()
        ''' % {'tempfile': repr(str(fn))}])
        with handler:
            logger.error('this is a really bad idea')
>       with fn.open() as rf:
            contents = rf.read().strip()

tests\test_more.py:92:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
envs\py34\lib\site-packages\py\_path\local.py:340: in open
    return py.error.checked_call(open, self.strpath, mode)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <py._error.ErrorMaker object at 0x0000000003662A90>
func = <built-in function open>
args = ('C:\\Users\\Smirnov\\AppData\\Local\\Temp\\pytest-63\\test_external_application_hand0\\tempfile', 'r')
kwargs = {}, __tracebackhide__ = False, cls = <class 'py.error.ENOENT'>
value = FileNotFoundError(2, 'No such file or directory')
tb = <traceback object at 0x00000000051C2BC8>, errno = 2

    def checked_call(self, func, *args, **kwargs):
        """ call a function and raise an errno-exception if applicable. """
        __tracebackhide__ = True
        try:
            return func(*args, **kwargs)
        except self.Error:
            raise
        except (OSError, EnvironmentError):
            cls, value, tb = sys.exc_info()
            if not hasattr(value, 'errno'):
                raise
            __tracebackhide__ = False
            errno = value.errno
            try:
                if not isinstance(value, WindowsError):
                    raise NameError
            except NameError:
                # we are not on Windows, or we got a proper OSError
                cls = self._geterrnoclass(errno)
            else:
                try:
                    cls = self._geterrnoclass(_winerrnomap[errno])
                except KeyError:
                    raise value
>           raise cls("%s%r" % (func.__name__, args))
E           py.error.ENOENT: [No such file or directory]: open('C:\\Users\\Smirnov\\AppData\\Local\\Temp\\pytest-63\\test_external_application_hand0\\tempfile', 'r')

..\envs\py34\lib\site-packages\py\_error.py:84: ENOENT
---------------------------- Captured stderr call -----------------------------
Traceback (most recent call last):
  File "logbook\handlers.py", line 214, in handle
    self.emit(record)
  File "logbook\more.py", line 277, in emit
    c = self._subprocess.Popen(args, stdin=stdin)
  File "envs\py34\lib\subprocess.py", line 858,
 in __init__
    restore_signals, start_new_session)
  File "envs\py34\lib\subprocess.py", line 1085, in _execute_child
    args = list2cmdline(args)
  File "envs\py34\lib\subprocess.py", line 663,
 in list2cmdline
    needquote = (" " in arg) or ("\t" in arg) or not arg
TypeError: Type str doesn't support the buffer API
Logged from file tests\test_more.py, line 91
```

**Comment by RazerM** [comment]
For `test_external_application_handler`, `ExternalApplicationHandler` encodes the arguments, but subprocess uses an internal function on Windows called `list2cmdline` ([source here](https://github.com/python/cpython/blob/c63e31aeafacdcdcae3b2df5df7f2607fdefffda/Lib/subprocess.py#L715-L782)) which expects a list of strings.

**Comment by RazerM** [comment]
@vmalloc This can be closed, #158 fixes these tests.

**Comment by vmalloc** [comment]
@RazerM awesome. Thanks!
